examples/tests/sdk_tests/resource/resource.test.w

```diff
@@ -0,0 +1,77 @@
+bring sim;
+
+inflight class CounterBackend impl sim.IResource {
+  var counter: num;
+
+  new() {
+    this.counter = 0;
+  }
+
+  pub onStart() {
+    // startup code
+  }
```

Comment:
We need to make sure there's a way to create/resolve tokens:

```suggestion
  pub onStart(ctx) {
    ctx.resolveToken("boom", value);
  }
```

Maybe this is a nice API for resolving tokens?

Also:
- how can you access things like the state directory of this resource?
- how does the resource emit logs (please include levels!)

(saying - let's try to make sure this API supports everything one needs in order to implement resources like a king).